config/buses/natss/100-natss.yaml

```diff
@@ -0,0 +1,106 @@
+apiVersion: v1
+data:
+  gnatsd.conf: |
+    # configuration file used to override default NATS server settings
+  stan.conf: |
+    # content of configuration file used to override default NATS Streaming server settings
+kind: ConfigMap
+metadata:
+  labels:
+    app: nats-streaming
+  name: nats-streaming
+  namespace: default
+---  
+apiVersion: v1
+kind: Service
+metadata:
+  name: nats-streaming-headless
+  namespace: default
+  labels:
+    app: nats-streaming
+spec:
+#  type: NodePort
+  type: ClusterIP
+#  clusterIP: None
+  ports:
+  - name: client
+    port: 4222
+    protocol: TCP
+    targetPort: client
+  selector:
+    app: nats-streaming
+  sessionAffinity: None
+  selector:
+    app: nats-streaming
+---    
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: nats-streaming
+  namespace: default
+  labels:
+    app: nats-streaming
+spec:
+  serviceName: nats-streaming-headless
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nats-streaming
+  serviceName: nats-streaming-headless
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "true"
+      labels:
+        app: nats-streaming
+    spec:
+      containers:
+        - name: nats-streaming
+          image: nats-streaming:0.11.0
+          imagePullPolicy: IfNotPresent
+          args:
+          - -D
+          - -SD
+          - --cluster_id=knative-nats-streaming
+          - --http_port=8222
+          - --max_age=24h
+          - --store=FILE
+          - --dir=/var/lib/nats-streaming/core-nats-streaming/$(POD_NAME)
+          - --port=4222
+          - --config=/etc/nats-streaming/core-nats-streaming/gnatsd.conf
+          - --stan_config=/etc/nats-streaming/core-nats-streaming/stan.conf
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+          ports:
+          - containerPort: 4222
+            name: client
+            protocol: TCP
+          - containerPort: 8222
+            name: monitoring
+            protocol: TCP            
+          volumeMounts:
+          - mountPath: /var/lib/nats-streaming/core-nats-streaming
+            name: datadir
+          - mountPath: /etc/nats-streaming/core-nats-streaming
+            name: config-volume
+          resources:
+            requests:
+              cpu: "100m"
+            limits:
+              memory: "32M"          
+      volumes:
+      - configMap:
+          name: nats-streaming
+        name: config-volume
+  volumeClaimTemplates:
+  - metadata:
+      name: datadir
+    spec:
+      accessModes:
+        - "ReadWriteOnce"
+      resources:
+        requests:
+          storage: "1Gi"
```

Comment:
Is this for the persistence of the messages? If so, is there a way to get a rough idea how this might be tweaked depending on message rates / sizes, etc.?

Reply:
Yes, this is for persistence, for all messages... We have to improve the strategy for the persistence layer in the future.